app/src/views/private/components/header-bar-actions/header-bar-actions.vue

```diff
@@ -34,8 +34,14 @@ export default defineComponent({
 	},
 	emits: ['toggle:sidebar'],
 	setup() {
+		const actionButtonsEl = ref<Element | null>(null);
 		const active = ref(false);
-		return { active };
+
+		const children = computed(() => {
+			return actionButtonsEl?.value?.children;
+		});
```

Comment:
Can we try checking this through "slots" in Vue, instead of relying on the DOM? 🤔

Reply:
You should be able to access the used slots from Vue's context, and check the length of default slot components instead

Reply:
I was trying to do so and I have looked again into it but I think I cannot rely on slots.

For example:
https://github.com/directus/directus/blob/1b053e6777f653e72c5ddf43795d426fe9703d53/app/src/modules/settings/routes/data-model/field-detail/field-detail.vue#L90-L100

---

https://github.com/directus/directus/blob/1b053e6777f653e72c5ddf43795d426fe9703d53/app/src/modules/settings/routes/data-model/field-detail/components/actions.vue#L2-L17

In this case we have a `div` and inside we have two `buttons`.

Calling `$slot.default()` returns the Vue instance of that slot and inside `children` we got the nested Vue instances of the components inside the template. In this case, the "leaf" of this nested structure is the component `setup-actions`, not the `v-button` I was expecting. 

I am not finding a way to do this other than accessing DOM.
Do you know any way to do this? Just point some direction and I will try to follow.

Reply:
I mean, I can add a `prop` or use `provide/inject` to achieve this behavior.
Should I follow this way?

Reply:
I have tried to look into this again but it seems very complex to achieve the same behaviour only with the slots.

We would need to recursive look into `scope.children.children...` and `scope.el.el.el...` until we find `children: null` or `el: null`. [related](https://forum.vuejs.org/t/how-to-determine-if-slot-is-empty-in-vue3/109154)

Should we follow this path, keep the PR or close it and maybe we will find a better solution for this in the future?

Reply:
> Should we follow this path, keep the PR or close it and maybe we will find a better solution for this in the future?

I think the latter is the better approach here. The solution outlined feels a little too hacky for my taste 👍🏻